profiles/views_test.py

```diff
@@ -292,3 +261,72 @@ def test_staff_sees_entire_profile(self):
         self.client.force_login(self.user1)
         resp = self.client.get(self.url2)
         assert resp.json() == ProfileSerializer().to_representation(profile)
+
+
+class ProfilePATCHTests(ProfileBaseTests):
+    """
+    Tests for profile PATCH
+    """
+    def test_patch_own_profile(self):
+        """
+        A user PATCHes their own profile
+        """
+        with mute_signals(post_save):
+            ProfileFactory.create(user=self.user1, filled_out=False, agreed_to_terms_of_service=False)
+        self.client.force_login(self.user1)
+
+        with mute_signals(post_save):
+            new_profile = ProfileFactory.create(filled_out=False)
+        new_profile.user.social_auth.create(
+            provider=EdxOrgOAuth2.name,
+            uid="{}_edx".format(new_profile.user.username)
+        )
+        patch_data = ProfileSerializer().to_representation(new_profile)
+
+        with patch(
+            'profiles.views.ProfileSerializer.__new__',
+            autospec=True,
+            side_effect=ProfileSerializer.__new__
+        ) as mocked:
+            resp = self.client.patch(self.url1, content_type="application/json", data=json.dumps(patch_data))
+        assert resp.status_code == 200
+        assert mocked.called
+
+        old_profile = Profile.objects.get(user__username=self.user1.username)
+        for key, value in patch_data.items():
+            field = ProfileSerializer().fields[key]
+
+            if isinstance(field, (ListSerializer, SerializerMethodField, ReadOnlyField)):
+                # these fields are readonly
+                continue
+            elif isinstance(field, DateField):
+                assert getattr(old_profile, key) == parse(value).date()
+            else:
+                assert getattr(old_profile, key) == value
+
+    def test_validate_filled_out(self):
```

Comment:
1. doesn't seem like an accurate name for this test case. maybe `test_filled_out_serializer`?
1. can you move the similar `profiles.views.ProfileSerializer` block of testing code above into its own test case?